Added SASS solutions, fixed typos, code formatting

drecali opened the pull request:
Added solutions to 3 guide articles. Also fixed typos and added code formatting for one of those articles. I kept the solutions and other edits as separate commits for clarity. Hope this helps. Please let me know if there's anything I can do to improve.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

drecali committed:
Added SASS solution and output CSS

drecali committed:
Fixed typo, added code formatting

The example code outputs for "start to end" and "start through end" were swapped. "1 to 3" does not include the end number, so should output "1 2", while "1 through 3" includes the end number, so should output "1 2 3"

drecali committed:
Added SASS solution and output CSS

drecali committed:
Added SASS solution

thecodingaviator reviewed (changes requested):
Pleae replace the solution with the full solution, i.r. the seed code with the changes you made for passing. Make it such that if the user copy-pastes the solution you give, the challenge passes

drecali committed:
Updated solution to full solution

Now includes all seed code, plus the changes needed to pass the tests, per @thecodingaviator [suggestion](https://github.com/freeCodeCamp/freeCodeCamp/pull/34792#pullrequestreview-210097979).

drecali committed:
Updated solution to full solution

Now includes all seed code, plus the changes needed to pass the tests, per @thecodingaviator suggestion https://github.com/freeCodeCamp/freeCodeCamp/pull/34792#pullrequestreview-210097979.

thecodingaviator commented on guide/english/certifications/front-end-libraries/sass/use-if-and-else-to-add-logic-to-your-styles/index.md:
```diff
@@ -3,8 +3,29 @@ title: Use @if and @else to Add Logic To Your Styles
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
-
-<!-- The article goes here, in GitHub-flavored Markdown. Feel free to add YouTube videos, images, and CodePen/JSBin embeds  -->
+## Solution
```
This solution is messed up, requires the HTML part too

thecodingaviator commented on guide/english/certifications/front-end-libraries/sass/use-for-to-create-a-sass-loop/index.md:
````diff
@@ -50,9 +50,50 @@ The `@for` loop might be useful when combined with CSS’ `:nth-*` pseudo-classe
+<p class="text-2">Hello</p>
+<p class="text-3">Hello</p>
+<p class="text-4">Hello</p>
+<p class="text-5">Hello</p>```
````
```suggestion
<p class="text-5">Hello</p>
```

thecodingaviator replied:
The three backticks would go on the next line

drecali committed:
Fixed Markdown syntax

Minor edit

drecali committed:
Updated solution to full solution

Now includes all seed code, plus the changes needed to pass the tests, per @thecodingaviator suggestion https://github.com/freeCodeCamp/freeCodeCamp/pull/34792#discussion_r263731937.

drecali commented:
@thecodingaviator thanks! I committed all the suggested changes.

drecali reviewed (commented):
Finished all the suggested changes. Thanks for the review!

thecodingaviator commented on guide/english/certifications/front-end-libraries/sass/use-if-and-else-to-add-logic-to-your-styles/index.md:
```diff
@@ -3,8 +3,32 @@ title: Use @if and @else to Add Logic To Your Styles
 ## Use @if and @else to Add Logic To Your Styles
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/certifications/front-end-libraries/sass/use-if-and-else-to-add-logic-to-your-styles/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+## Solution
```
Your given solution resolves to this:
![image](https://user-images.githubusercontent.com/34807532/54029722-6a5c9980-41cf-11e9-8558-063829feb7ae.png)

thecodingaviator replied:
@Bouncey / @ValeraS Is this a problem on our end?

RandellDawson replied:
@thecodingaviator I copied/pasted the solution provided in this PR and used it in current production and beta site and it passed all tests.  I also tested locally using current master and it passes with no errors.  How were you testing?

thecodingaviator replied:
I was testing on latest beta, seems fixed now, may have been a problem on my end

thecodingaviator reviewed (approved):
LGTM! 👍

thecodingaviator commented:
@drecali Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝
